tools/depends/download-files.include

```diff
@@ -0,0 +1,33 @@
+#
+HASH_FOUND = yes
+ifneq ($(SHA512),)
+  HASH_TYPE = sha512
+  HASH_TOOL ?= $(SHA512SUM)
+else ifneq ($(SHA256),)
+  HASH_TYPE = sha256
+  HASH_TOOL ?= $(SHA256SUM)
+else
+  HASH_FOUND = no
+endif
+ifeq ($(NATIVE_OS), osx)
+  HASH_TOOL = $(SHASUM)
+endif
+
+HASH_SUM = $($(shell echo $(HASH_TYPE) | tr  '[:lower:]' '[:upper:]'))
+
+.PHONY: $(TARBALLS_LOCATION)/$(ARCHIVE).$(HASH_TYPE)
+all: $(TARBALLS_LOCATION)/$(ARCHIVE).$(HASH_TYPE)
+
+$(TARBALLS_LOCATION)/$(ARCHIVE):
+	cd $(TARBALLS_LOCATION); $(RETRIEVE_TOOL) $(RETRIEVE_TOOL_FLAGS) $(BASE_URL)/$(ARCHIVE)
+
+$(TARBALLS_LOCATION)/$(ARCHIVE).$(HASH_TYPE): $(TARBALLS_LOCATION)/$(ARCHIVE)
+ifeq ($(HASH_FOUND),no)
+	cd $(TARBALLS_LOCATION); $(SHA512SUM) $(ARCHIVE) > $(ARCHIVE).sha512
+	SHA512=$$(cat $(TARBALLS_LOCATION)/$(ARCHIVE).sha512 | cut -d ' ' -f 1) ; sed -i "s/#SHA512#/SHA512=$${SHA512}/" Makefile
```

Comment:
Another mac oddity regarding the sed command. We currently work around it elsewhere with the following
https://github.com/xbmc/xbmc/blob/22e424739448632d9db0ee80089aef72d95cfab1/tools/depends/target/pythonmodule-pil/Makefile#L36-L39
https://github.com/xbmc/xbmc/blob/22e424739448632d9db0ee80089aef72d95cfab1/tools/depends/target/pythonmodule-pil/Makefile#L72-L75

Essentially -i on mac sed requires either the empty '' to do an insert with no backup or something that will append to filename for a backup.
You may see in other instances of sed use there is sed -ie used.